### PR TITLE
Core/Player Drop deprecated code and fix bug where a player can have …

### LINKED ISSUE #226: Core/Quest: Quests related to professions/specializations must be removed if the profession/specialization is forgetten
Core revision: Trinity Core 10868 Operating System: Win32, little-endian
using TDB UP36 (doesn't really matter, it's a core issue)

When you get a profession specialization (ie. armorsmith as blacksmith spec) and you try to change to another profession specialization (weaponsmith) by either unlearning it and learning in the normal, blizzlike way or using .cast (spell to unlearn that specialization) after relogging, you have both profession specializations (ie. armorsmith and weaponsmith).
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/66453-profession-specializations?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Asterc**
…more than one profession specialization.

Fix by @elecyb

Closes #226

### CODE REVIEW

**Comment by tkrokli** [comment]
Nice cleanup, it goes well with the general wish for making Player.cpp smaller in lenght. :+1:

**Comment by Keader** [comment]
Important comment about this https://github.com/TrinityCore/TrinityCore/pull/16152#issuecomment-168849299

**Comment by jackpoz** [comment]
Could you describe more in detail what the changes in this PR do ?

**Comment by elecyb** [comment]
@jackpoz Read my prev comment https://github.com/TrinityCore/TrinityCore/pull/16152#issuecomment-168849299
The code deleted is obsolete and nonsense because specializations spells can be re-learned from the profession trainers or from the book "Soothsaying for dummies" for engineering. There are scripts that handle those cases.

**Comment by jackpoz** [comment]
So both cases https://github.com/TrinityCore/TrinityCore/issues/226#issuecomment-8810386 and https://github.com/TrinityCore/TrinityCore/issues/226#issuecomment-32110368 have been tested already ingame and they work correctly now ?

**Comment by Asterc** [comment]
@jackpoz In this PR, we are avoiding relearning skills that have been previously unlearned.
It will prevent a wrongly relearn of the spells listed above, only
```
2656    Smelting
9787    Weaponsmith
9788    Armorsmith
10656   Dragonscale Leatherworking
10658   Elemental Leatherworking
10660   Tribal Leatherworking
17039   Master Swordsmith
17040   Master Hammersmith
17041   Master Axesmith
20219   Gnomish Engineer
20222   Goblin Engineer
26797   Spellfire Tailoring
26798   Mooncloth Tailoring
26801   Shadoweave Tailoring
28672   Transmutation Master
28675   Potion Master
28677   Elixir Master
```

About the mentioned issue https://github.com/TrinityCore/TrinityCore/issues/226#issuecomment-8810386 doesn't belong here because the specialization is learned without having the profession itself. So, there's no relearn, which is what we are handling here.

https://github.com/TrinityCore/TrinityCore/issues/226#issuecomment-32110368 isn't been affected by this pr too

**Comment by jackpoz** [comment]
another issue will have to be opened then after #226 is marked as closed

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -22602,46 +22602,16 @@ void Player::LearnQuestRewardedSpells(Quest const* quest)
...
-                    if (itrInfo->Effects[0].Effect != SPELL_EFFECT_TRADE_SKILL || itrInfo->Effects[1].Effect != 0)
-                        continue;
-
-                    // compare same chain spells
-                    if (sSpellMgr->IsSpellRequiringSpell(itr->first, profSpell))
-                        return;
-                }
-            }
-        }
+       // profession specialization can be re-learned from npc
```
Comment (killradio) [inline src/server/game/Entities/Player/Player.cpp]: > Note, however, that you do not unlearn your speciality, if you have one. If you later re-acquire an unlearnt profession for which you had a speciality, you will still have the speciality.

From wiki.

**Pull request merged**